ts/main.ts

```diff
@@ -153,3 +189,29 @@ function handleClick(event: Event): void {
 
 if (!$newButton) throw new Error('$newButton not found!');
 $newButton.addEventListener('click', () => viewSwap('entry-form'));
+
+if (!$ulForEntries) throw new Error('$ulForEntries not found!');
+$ulForEntries.addEventListener('click', HandleUlClick);
+
+function HandleUlClick(event: Event): void {
+  const eventTarget = event.target as HTMLElement;
+  if (eventTarget.matches('.fa-pencil')) {
+    viewSwap('entry-form');
+    const match = data.entries.find(
+      (v) => v.entryId === Number(eventTarget.dataset.entryId),
+    );
+    if (match) {
+      data.editing = match;
```

Comment:
Rather than checking if `match` is truthy and nesting everything in your condition code block, I would say make this a guard clause and return if `match` is _not_ truthy so that the code below it is easier to read.